### PR TITLE
The player can now change the spaceship's appearance by pressing 'shift'

### PR DESCRIPTION
**Pull request by pmuren**
Check out the hot new feature!

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -87,6 +92,15 @@ class Hero extends Pixi.Sprite {
...
             this.position.x += this.speed * delta
         }
+
+        // This will change the sprite's texture
+        if(Keyb.isJustDown("<shift>")) {
+            this.changeTexture()
+        }
+    }
+    changeTexture() {
+        this.textureIndex = (this.textureIndex + 1) % spaceshipTextures.length
```
Comment (ehgoodenough) [inline source/index.js:102]: Nice use of modulo! :D

**Comment by ehgoodenough** [comment]
What a killer feature!

File: source/index.js [hunk 2]
```diff
@@ -56,6 +56,10 @@ document.body.appendChild(renderer.view)
 // load these textures from our `images` directory.
 var RED_TEXTURE = Pixi.Texture.fromImage(require("images/red-starship.png"))
 var BLUE_TEXTURE = Pixi.Texture.fromImage(require("images/blue-starship.png"))
+var GREEN_TEXTURE = Pixi.Texture.fromImage(require("images/green-starship.png"))
+var YELLOW_TEXTURE = Pixi.Texture.fromImage(require("images/yellow-starship.png"))
+
+var spaceshipTextures = [RED_TEXTURE, BLUE_TEXTURE, GREEN_TEXTURE, YELLOW_TEXTURE]
```
Comment (ehgoodenough) [inline source/index.js:62]: Maybe we could use this like a queue, so whatever texture is at the zero index is what we assign to the `this.texture`.

``` js
    // Remove the first element and put it at the end.
    spaceshipTextures.push(spaceshipTextures.shift())
    this.texture = spaceshipTextures[0]
```

**Comment by ehgoodenough** [comment]
Could we get a playable link for this PR?

**Comment by pmuren** [comment]
How do I make that happen?

**Comment by ehgoodenough** [comment]
I added a bash script called "share.sh" in the root of the project. If you run it with:

```
$ ./share.sh pr2
```

Then the code will be compiled and uploaded for you! :D

**Comment by pmuren** [comment]
I did so, but where does it upload the playable? Thanks for the tutorial, btw!

**Comment by ehgoodenough** [comment]
If it worked, it should upload it to: https://ehgoodenough.github.io/ld36/pr2. It'll usually print that out for you, so you can copy it into your pull request! :]

It looks like it might not have worked? Did you get some error messages?

**Comment by ehgoodenough** [comment]
(Maybe my script messed up? Wouldn't be the first time.)

**Comment by pmuren** [comment]
Well, I messed up my credentials. Then it asked me for them on the command line, and I entered them correctly. Now the script is erroring out. My b.

**Comment by ehgoodenough** [comment]
Does running it again fix it? If not, send me a screenshot of the errors! :D

**Comment by pmuren** [comment]
Unpacking objects: 100% (40/40), done.
Checking connectivity... done.
[04:10:PM] Building (2.55s)(1.97MB)
warning: LF will be replaced by CRLF in pr2/index.html.
The file will have its original line endings in your working directory.
warning: LF will be replaced by CRLF in pr2/index.js.
The file will have its original line endings in your working directory.

**\* Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for <(null)>) not allowed

**Comment by pmuren** [comment]
After updating my git config with my user name:
he file will have its original line endings in your working directory.
warning: LF will be replaced by CRLF in pr2/index.js.
The file will have its original line endings in your working directory.

**\* Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'Bland@DESKTOP-NA8ICUT.(none)')

**Comment by ehgoodenough** [comment]
It's still giving the error? Even after git configing?

**Comment by ehgoodenough** [comment]
What's in ~/.gitconfig at the moment?

**Comment by pmuren** [comment]
Fixed by updating my git config email. ;)))

**Comment by ehgoodenough** [comment]
Nice! 👍

**Comment by pmuren** [comment]
I don't think your script expected the null email from when I failed the credential entry.

**Comment by ehgoodenough** [comment]
Nope, I don't think it did 0.o

I need better error handling :p

**Comment by pmuren** [comment]
I've noticed that Keyb.isJustDown does not register the very first press of the key. Any idea why?

**Comment by ehgoodenough** [comment]
Good catch! I bet it's because the internal map that it keeps for which keys are down and which keys are up hasn't been initialized for that key yet. An oversight on Keyb's part.

**Comment by ehgoodenough** [comment]
You should make an issue at: https://github.com/ehgoodenough/keyb/issues

**Comment by ehgoodenough** [comment]
So did the build get uploaded? Share me the link! :D

**Comment by pmuren** [comment]
https://ehgoodenough.github.io/ld36/pr2
This link is working for me!

**Comment by ehgoodenough** [comment]
Excellent! It works for me, too! Let's get this merged!

**Comment by pmuren** [comment]
Should I not be merging gh-pages into master?

**Comment by ehgoodenough** [comment]
Yeah, don't merge gh-pages into master. All of our branches, like ship-texture-swap and master, hold our actual source code. The stuff we push to gh-pages is the compiled builds. When we merge this pull request, I'm merging your source code from the ship-texture-swap branch to the master branch.

**Comment by ehgoodenough** [comment]
So whenever you start writing a new feature for a new pull request, start by pulling from master (and getting all the latest source code) then make a new branch, and start committing! We only push to gh-pages using that `share.sh` script (assuming it's working...)

**Comment by pmuren** [comment]
Cool, cool. So is it okay if I delete the pr1 folder from my local repo?

**Comment by ehgoodenough** [comment]
Yeah, totally! Thanks!

**Pull request merged**